rmw_zenoh_cpp/src/detail/rmw_subscription_data.cpp

```diff
@@ -173,151 +173,62 @@ bool SubscriptionData::init()
 
   sess_ = context_impl->session();
 
+  auto adv_sub_opts = zenoh::ext::SessionExt::AdvancedSubscriberOptions::create_default();
+
   // Instantiate the subscription with suitable options depending on the
   // adapted_qos_profile.
   // TODO(Yadunund): Rely on a separate function to return the sub
   // as we start supporting more qos settings.
   if (entity_->topic_info()->qos_.durability == RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL) {
-    zenoh::ext::SessionExt::QueryingSubscriberOptions sub_options =
-      zenoh::ext::SessionExt::QueryingSubscriberOptions::create_default();
-    const std::string selector = "*/" + entity_->topic_info()->topic_keyexpr_;
-    zenoh::KeyExpr selector_ke(selector);
-    sub_options.query_keyexpr = std::move(selector_ke);
-    // Tell the PublicationCache's Queryable that the query accepts any key expression as a reply.
-    // By default a query accepts only replies that matches its query selector.
-    // This allows us to selectively query certain PublicationCaches when defining the
-    // set_querying_subscriber_callback below.
-    sub_options.query_accept_replies = ZC_REPLY_KEYEXPR_ANY;
-    // As this initial query is now using a "*", the query target is not COMPLETE.
-    sub_options.query_target = Z_QUERY_TARGET_ALL;
-    // We set consolidation to none as we need to receive transient local messages
-    // from a number of publishers. Eg: To receive TF data published over /tf_static
-    // by various publishers.
-    sub_options.query_consolidation =
-      zenoh::QueryConsolidation(zenoh::ConsolidationMode::Z_CONSOLIDATION_MODE_NONE);
```

Comment:
And it does not look like there is a way to set these parameters anymore. Can you confirm that the behavior with the AdvancedSubscriber will match the expectations of the old implementation?

Reply:
Yes. They're preset in the same way https://github.com/eclipse-zenoh/zenoh/blob/26d67de043efe327befd0bbfb29c76a5157f1baf/zenoh-ext/src/advanced_subscriber.rs?plain=1#L615.